File: OpenCV.Algorithmic.Solution/chapter-02/2.4/imgToArray.py
# -*- coding: utf-8 -*-
import sys
import cv2
import numpy as np

# 主函数
if __name__ == "__main__":
    # 输入图像矩阵，转换为array
    if len(sys.argv) > 1:
        img = cv2.imread(sys.argv[1], cv2.IMREAD_GRAYSCALE)
    else:
        print("Usage: python imgToArray.py imageFile")

    # 显示图像
    cv2.imshow("img", img)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

File: OpenCV.Algorithmic.Solution/chapter-02/2.4/imgToArray.cpp
#include <string>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(int argc, char *argv[])
{
    // 输入图像矩阵
    Mat img = imread(argv[1], IMREAD_GRAYSCALE);
    if (img.empty())
        return -1;

    // 定义显示原图的窗口
    string winname = "原图";
    namedWindow(winname, WINDOW_AUTOSIZE);

    // 显示图像
    imshow(winname, img);
    waitKey(0);

    return 0;
}
